draft-saml-ipsie-sl1-profile.md

```diff
@@ -0,0 +1,382 @@
+- Credentials not intended for handling by end-users shall be created with at least 128 bits of entropy such that an attacker correctly guessing the value is computationally infeasible.
+- All key material shall be generated using FIPS 140-2 Level 1 validated random bit generators or higher.
+- Assertion signing keys shall be distinct from keys used for other purposes.
+
+### 3.3. SAML Requirements
+
+In the following, a profile of the following technologies is defined:
+
+- SAML 2.0 Core [SAML2Core]
+- SAML 2.0 Profiles [SAML2Profiles]
+- SAML 2.0 Metadata [SAML2Metadata]
+- SAML 2.0 Bindings [SAML2Bindings]
+
+#### 3.3.1. Requirements for Identity Providers
+
+Identity Providers:
+
+- shall distribute metadata as specified in [SAML2Metadata] through a protected channel with integrity protection;
+- shall only support pre-registered Service Providers with established trust relationships;
+- shall authenticate Service Providers using X.509 certificates or another strong authentication method;
+- shall sign all assertions using XML Signature [XMLDSig];
+- shall encrypt all assertions containing personally identifiable information (PII) or sensitive attributes using XML Encryption [XMLEnc] with AES-GCM with a minimum key length of 128 bits;
+- shall not expose open redirectors;
+- shall only accept entity IDs registered in metadata as audience values;
+- shall issue assertions with a maximum lifetime of 5 minutes;
+- shall require Service Providers to be preregistered, and shall not support dynamic registration of Service Providers;
+- shall support assertion revocation or status verification mechanisms (such as OCSP for certificate checking);
+- shall implement key rotation procedures with a maximum key lifetime of 2 years for signing keys.
+
+SAML Assertions issued by Identity Providers:
+
+- shall contain the Service Provider's entity ID as the audience value;
+- shall contain an AuthnContext element that identifies the authentication context class that the authentication performed satisfied;
+- shall include an AuthnContextClassRef indicating the authentication method used, which must reflect multi-factor authentication per NIST 800-63-4 AAL2 requirements;
+- shall indicate the expected lifetime of the Service Provider session in the SessionNotOnOrAfter attribute;
+- shall include NotBefore and NotOnOrAfter conditions with a maximum validity period of 5 minutes;
+- shall include a unique identifier for the assertion that is never reused;
+- shall include a SubjectConfirmation element that implements holder-of-key binding when assertions contain sensitive information or authorize high-value transactions;
+- shall include attribute statements that use pairwise pseudonymous identifiers for privacy protection when appropriate;
+- shall minimize disclosure of personal information by only including required attributes in assertions.
+
+For the Web Browser SSO Profile, Identity Providers:
```

Comment:
Missing requirements for processing AuthnRequests such as validating signed request if we want to require signed requests which you have below for the SP